src/main/java/wooteco/subway/domain/line/section/rule/SectionDeleteRule/MatchedSection.java

```diff
@@ -0,0 +1,51 @@
+package wooteco.subway.domain.line.section.rule.SectionDeleteRule;
+
+import wooteco.subway.domain.line.section.Section;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+public class MatchedSection {
+
+    private final Section sectionWithStationIdAsUp;
+    private final Section sectionWithStationIdAsDown;
+
+    public MatchedSection(List<Section> sections, Long stationId) {
+        this.sectionWithStationIdAsUp = findMatchedSectionWithUpSection(sections, stationId).orElse(null);
+        this.sectionWithStationIdAsDown = findMatchedSectionWithDownSection(sections, stationId).orElse(null);
+    }
+
+    private Optional<Section> findMatchedSectionWithUpSection(List<Section> sections, Long stationId) {
+        return sections.stream()
+                .filter(section -> Objects.equals(section.getUpStationId(), stationId))
```

Comment:
객체에게 메세지를 보내주세요 :) 
`Objects.equals` 사용한 곳이 많아 보이는데, 메세지 보내는 방식으로 수정 가능한지 살펴보면 좋을 것 같아요 :) 
```suggestion
                .filter(section ->section.isEqualsUpStationId(stationId))
```

Reply:
오.... 객체지향의 가장 기본 원칙을 놓쳤네요..... 감사합니다ㅠㅠ!!